tests/system_tests/test_api_endpoints.py

```diff
@@ -0,0 +1,190 @@
+import time
+from typing import Any
+
+import pytest
+import requests
+from fastapi import status
+from pydantic import TypeAdapter
+
+from blueapi.client.rest import BlueapiRestClient, BlueskyRemoteControlError
+from blueapi.service.interface import get_devices, get_plans
+from blueapi.service.model import (
+    DeviceResponse,
+    EnvironmentResponse,
+    PlanResponse,
+    TasksListResponse,
+    WorkerTask,
+)
+from blueapi.worker.event import TaskStatusEnum, WorkerState
+from blueapi.worker.task import Task
+from blueapi.worker.task_worker import TrackableTask
+
+_SIMPLE_TASK = Task(name="sleep", params={"time": 0.0})
+_LONG_TASK = Task(name="sleep", params={"time": 1.0})
+
+rest = BlueapiRestClient()
+
+
+def get_response(
+    url: str, BaseModel: Any, status_code: int = status.HTTP_200_OK
+) -> Any:
+    get_response = requests.get(url)
+    model = TypeAdapter(BaseModel).validate_python(get_response.json())
+    assert get_response.status_code == status_code
+    return model
+
+
+def test_get_plans():
+    assert rest.get_plans() == PlanResponse(plans=get_plans())
+
+
+def test_get_plans_by_name():
+    for plan in get_plans():
+        assert rest.get_plan(plan.name) == plan
+
+
+def test_get_non_existant_plan():
+    with pytest.raises(KeyError) as exception:
+        rest.get_plan("Not exists")
+    assert exception.value.args[0] == ("{'detail': 'Item not found'}")
+
+
+def test_get_devices():
+    assert rest.get_devices() == DeviceResponse(devices=get_devices())
+
+
+def test_get_device_by_name():
+    for device in get_devices():
+        assert rest.get_device(device.name) == device
+
+
+def test_get_non_existant_device():
+    with pytest.raises(KeyError) as exception:
+        assert rest.get_device("Not exists")
+    assert exception.value.args[0] == ("{'detail': 'Item not found'}")
+
+
+def test_create_task_and_delete_task_by_id():
```

Comment:
Could: This is good but could be split into two tests, `create_task` and `create_and_delete_task`. That way, if there is a problem, it's clearer where it is.

Reply:
To ensure a clean environment following the completion of a task, it is essential to revert any changes made during the process. For instance, if a test involves inserting data into a database, the appropriate approach would be to either remove the inserted data at the conclusion of the test or to instantiate a new database for each test. In this case, I have opted for the former approach, ensuring that the inserted data is deleted post-test to maintain a clean and consistent environment.